imagetest/test_suites/security/image_security_test.go

```
@@ -0,0 +1,327 @@
+	maxInterval                  = 7
+	minInterval                  = 1
+	sysctlBase                   = "/proc/sys/"
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	if *runtest {
+		os.Exit(m.Run())
+	} else {
+		os.Exit(0)
+	}
+}
+
+// TestKernelSecuritySettings Checks that the given parameter has the given value in sysctl.
+func TestKernelSecuritySettings(t *testing.T) {
+	for key, expect := range securitySettingMap {
+		v, err := sysctlGet(key)
+		if err != nil {
+			t.Fatalf("failed getting key %s", key)
+		}
+		value, err := strconv.Atoi(v)
+		if err != nil {
+			t.Fatalf("failed convert config %s value %s to int", key, v)
+		}
+		if expect != value {
+			t.Fatalf("expected %s = %d. found %d.", key, expect, value)
+		}
+	}
+}
+
+// TestAutomaticUpdates Check automatic security updates are installed and enabled.
+func TestAutomaticUpdates(t *testing.T) {
+	image, err := utils.GetMetadata("image")
+	if err != nil {
+		t.Fatalf("couldn't get image from metadata")
+	}
+
+	switch {
+	case strings.Contains(image, "rhel-8"), strings.Contains(image, "centos-8"),
+		strings.Contains(image, "centos-stream"), strings.Contains(image, "almalinux"):
+		if err := verifyServiceEnabled("dnf-automatic.timer"); err != nil {
+			t.Fatal(err)
+		}
+	case strings.Contains(image, "rhel-7"), strings.Contains(image, "centos-7"):
+		if err := verifyServiceEnabled("yum-cron"); err != nil {
+			t.Fatal(err)
+		}
+	case strings.Contains(image, "debian") || strings.Contains(image, "ubuntu"):
+		if err := verifyUpgradesPackageInstalled(); err != nil {
+			t.Fatal(err)
+		}
+		// Check that the security packages are marked for automatic update
+		// https://wiki.debian.org/UnattendedUpgrades
+		// https://help.ubuntu.com/community/AutomaticSecurityUpdates
+		if err := verifySecurityUpgradeEnabled(image); err != nil {
+			t.Fatal(err)
+		}
+		if err := verifyAutomaticUpdate(image); err != nil {
+			t.Fatal(err)
+		}
+	default:
+		t.Skipf("skipping test on image %s", image)
```

i think we should fail on the default - so when we introduce a new image, it will fail (alerting us to add support) rather than skipping and appearing to pass.